### PR TITLE
107 edit profile settings

### LINKED ISSUE #107: Edit profile settings
Add ability for user. Change user settings (Theme, Language)

### PR DESCRIPTION
**Pull request by Vova-Romaniuk**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/users/maryanchuk04/projects/3)

## Code reviewers

- [ ] @maryanchuk04 
- [ ] @Serhiy-dotcom 

### Second Level Review

- [ ] @Varvara-Maria 

## Summary of issue

Create settings page with possibility selected (theme interface and language etc.)

## Summary of change

Created all the necessary components for perform tasks

## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: StudCity.Frontend/src/pages/Profile/EditProfile/ProfileSettings/index.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+const Accord = ({ children, isExpanded = false }) => {
+	const [expanded, setExpanded] = useState(isExpanded);
+
+	return (
+		<div className="border-2 border-borderGray py-2 px-5 rounded-lg first:mt-5 last:mb-5">
+			{expanded ?
+				React.Children.map(children, (child, index) => {
+					if (index === 0) {
+						return React.cloneElement(child, { setExpanded: setExpanded, expanded: expanded });
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/Profile/EditProfile/ProfileSettings/index.jsx]: I see what you are doing here, but is it possible to omit using React native methods? I would suggest to declare HeaderAccord directrly inside Accord component, and just pass HeaderAccord text through props in Accord component, that's one option that I see how to omit this kind of logic

**Review by Serhiy-dotcom — COMMENTED** [review]
Please check comments
PS: you have one merge conflict

File: StudCity.Frontend/src/UI/DatePicker/CustomHeader.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ const CustomHeader = ({
 	prevMonthButtonDisabled,
 	nextMonthButtonDisabled,
 }) => {
-	const buttonStyles = "border-2 bg-white border-[#D1D7D4] rounded-full px-2 py-1";
+	const buttonStyles = "border-2 bg-white border-borderGray rounded-full px-2 py-1";
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/UI/DatePicker/CustomHeader.jsx]: Rename color name pls

File: StudCity.Frontend/src/components/Scroller/index.jsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React from 'react'
 
 const Scroller = ({ children }) => {
 	return (
-		<div className='h-full w-full overflow-y-auto'>
+		<div className='h-full w-full flex overflow-y-auto'>
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/components/Scroller/index.jsx]: Are you sure that this does not affect other components? You might want to add a className property to this component

File: StudCity.Frontend/src/pages/Profile/EditProfile/ProfileSettings/index.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import SelectField from "../../../../UI/fields/SelectField";
+import ToggleContainerText from "../../../../UI/ToggleContainerText";
+import Button from "../../../../UI/Button";
+
+export default function ProfileSettings() {
+	return (
+		<div className="w-10/12 min-h-fit max-h-full mx-auto ml-6 flex flex-col gap-5">
+			<Accord isExpanded={true}>
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/pages/Profile/EditProfile/ProfileSettings/index.jsx]: I think it's a bad idea to do it that way!

Why not make components if they were called Accordion, in which the isExpanded title and children properties would wake up, this component should be more reusable, and we can transfer your toggles to another component, they can be useful

File: StudCity.Frontend/src/pages/Profile/EditProfile/ProfileSettings/index.jsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+
+const Accord = ({ children, isExpanded = false }) => {
+	const [expanded, setExpanded] = useState(isExpanded);
+
+	return (
+		<div className="border-2 border-borderGray py-2 px-5 rounded-lg first:mt-5 last:mb-5">
+			{expanded ?
+				React.Children.map(children, (child, index) => {
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/pages/Profile/EditProfile/ProfileSettings/index.jsx]: KISS!!!

**Review by maryanchuk04 — CHANGES_REQUESTED** [review]

**Comment by maryanchuk04** [comment]
@Vova-Romaniuk Please review all comments

**Review by Serhiy-dotcom — APPROVED** [review]
Nice work!

**Review by maryanchuk04 — APPROVED** [review]
Great job!

**Pull request merged**